### PR TITLE
Clarify doc for ingressclass name in k8s 1.18+

### PR DESCRIPTION
**Pull request by tomMoulard**
### What does this PR do?

Explaining the ingressclass name filter.

### Motivation

Update the Doc

### More

~~- [ ] Added/updated tests~~
- [X] Added/updated documentation

### CODE REVIEW

File: docs/content/providers/kubernetes-ingress.md [hunk 1]
```diff
@@ -224,7 +224,9 @@ Otherwise, Ingresses missing the annotation, having an empty value, or the value
 
     If the Kubernetes cluster version is 1.18+,
     the new `IngressClass` resource can be leveraged to identify Ingress objects that should be processed.
-    In that case, Traefik will look for an `IngressClass` in the cluster with the controller value equal to *traefik.io/ingress-controller*. 
+    In that case, Traefik will look for an `IngressClass` in the cluster with the controller value equal to *traefik.io/ingress-controller*.
+
+    In addition to the controller value matching mechanism, the property `ingressClass` (if set) will be used to select IngressClasses objects by their name by applying a strict matching.
```
Comment (rtribotte) [inline docs/content/providers/kubernetes-ingress.md:229]:
```suggestion
    In addition to the controller value matching mechanism, the property `ingressClass` (if set) will be used to select IngressClasses objects by their name, applying a strict matching.
```

**Review by rtribotte — APPROVED** [review]
LGTM 👌

**Review by SantoDE — APPROVED** [review]
LGTM 👼

**Review by kevinpollet — APPROVED** [review]
LGTM

**Pull request merged**